subsys/dfu/boot/mcuboot.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2017 Nordic Semiconductor ASA
+ * Copyright (c) 2016 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __MCUBOOT_H__
+#define __MCUBOOT_H__
+
+#include "boot/mcuboot_constraints.h"
+
+/**
+ * @brief Marks the image in slot 0 as confirmed. The system will continue
+ * booting into the image in slot 0 until told to boot from a different slot.
+ *
+ * This call is expected to be used by the application running on trial.
+ *
+ * @return 0 on success, negative errno code on fail.
+ */
+int boot_img_confirmed_write(void);
```

Comment:
Just a nit: "boot_img_confirmed_write" is "subject object verb" order, and "boot_request_upgrade" is "subject verb object" order.

Since "subject verb object" is the usual order in English, how about renaming this to "boot_write_img_confirmed", and similarly using "boot_erase_img_bank" below? To be clear, I'd be fine with doing that as a cleanup patch that gets sent later on.